etna/pipeline/hierarchical_pipeline.py

```diff
@@ -61,18 +59,23 @@ def fit(self, ts: TSDataset) -> "HierarchicalPipeline":
         Parameters
         ----------
         ts:
-            Dataset with hierarchical timeseries data
+            Dataset with hierarchical timeseries data.
+        save_ts:
+            Will ``ts`` be saved in the pipeline during ``fit``.
 
         Returns
         -------
         :
             Fitted HierarchicalPipeline instance
         """
-        self._fit_ts = deepcopy(ts)
+        if save_ts:
+            self.ts = ts
 
         self.reconciliator.fit(ts=ts)
         ts = self.reconciliator.aggregate(ts=ts)
-        super().fit(ts=ts)
+        super().fit(ts=ts, save_ts=False)
```

Comment:
A little bit tricky because `ts` is actually modified in pipeline

Reply:
We have such a behavior in any other pipeline.